src/types.ts

```diff
@@ -0,0 +1,25 @@
+export type ScopeStr = 'device' | 'local' | 'private' | 'public';
+
+export type Incoming = {
+  scope: ScopeStr | Array<ScopeStr>;
+  transform: 'shs' | 'noauth';
+  port?: number;
+  host?: string;
+};
+
+export type Outgoing = {
+  scope?: undefined;
+  transform: 'shs' | 'noauth';
+  port?: undefined;
+  host?: undefined;
+};
```

Comment:
This specifies an object shape that requires a `transform` (which is either the string `shs` or the string `noauth`, nothing else), but also:

- forbids the keys `scope`, `port`, `host` to be present in this object shape